base/subarray.jl

```diff
@@ -316,7 +316,7 @@ reindex(a, b) = a[b]
 reindex(a::UnitRange, b::UnitRange{Int}) = range(oftype(first(a), first(a)+first(b)-1), length(b))
 reindex(a::UnitRange, b::StepRange{Int}) = range(oftype(first(a), first(a)+first(b)-1), step(b), length(b))
 reindex(a::StepRange, b::Range{Int}) = range(oftype(first(a), first(a)+(first(b)-1)*step(a)), step(a)*step(b), length(b))
-reindex(a, b::Int) = unsafe_getindex(a, b)
+reindex(a, b::Int) = (@inbounds r = a[b]; r)
```

Comment:
We could probably get rid of `reindex` altogether. The only reason we had it before is that it was impossible to turn off bounds checks on `(3:6)[1:2]`.